### PR TITLE
Update diplomat

### PR DESCRIPTION
**Pull request by sffc**
PTAL at the changes.  There are some actual diffs with the new alignment change.  But, `cargo make wasm-test-release` passes.

### CODE REVIEW

File: ffi/diplomat/wasm/lib/api.mjs [hunk 1]
```diff
@@ -785,20 +800,21 @@ export class ICU4XPluralOperands {
 
   static create(s) {
     let s_diplomat_bytes = (new TextEncoder()).encode(s);
-    let s_diplomat_ptr = wasm.diplomat_alloc(s_diplomat_bytes.length);
+    let s_diplomat_ptr = wasm.diplomat_alloc(s_diplomat_bytes.length, 1);
     let s_diplomat_buf = new Uint8Array(wasm.memory.buffer, s_diplomat_ptr, s_diplomat_bytes.length);
     s_diplomat_buf.set(s_diplomat_bytes, 0);
     const diplomat_out = (() => {
-      const diplomat_receive_buffer = wasm.diplomat_alloc(25);
+      const diplomat_receive_buffer = wasm.diplomat_alloc(37, 8);
```
Comment (sffc) [inline ffi/diplomat/wasm/lib/api.mjs:807]: Is this correct?

File: ffi/diplomat/wasm/lib/api.mjs [hunk 2]
```diff
@@ -807,23 +823,23 @@ export class ICU4XPluralOperands {
   }
 
   get v() {
-    return (new Uint32Array(wasm.memory.buffer, this.underlying + 4, 1))[0];
+    return (new Uint32Array(wasm.memory.buffer, this.underlying + 8, 1))[0];
```
Comment (sffc) [inline ffi/diplomat/wasm/lib/api.mjs:826]: Are these changes to the offsets correct?

File: ffi/diplomat/wasm/lib/api.mjs [hunk 3]
```diff
@@ -75,7 +75,7 @@ export class ICU4XCreatePluralOperandsResult {
   }
 
   get success() {
-    return (new Uint8Array(wasm.memory.buffer, this.underlying + 24, 1))[0] == 1;
+    return (new Uint8Array(wasm.memory.buffer, this.underlying + 36, 1))[0] == 1;
```
Comment (sffc) [inline ffi/diplomat/wasm/lib/api.mjs:78]: Is this change to the offset correct?

**Comment by Manishearth** [comment]
I have a hard time being sure of the offset values either before or after -- do the tests pass?

**Review by Manishearth — APPROVED** [review]
r=me as long as wasm tests pass

File: ffi/diplomat/wasm/lib/api.mjs (hunk 2 above)
Comment (sffc) [reply ffi/diplomat/wasm/lib/api.mjs:826]: OK, so the difference appears to be that ICU4XPluralOperands contains some `u64` values.  Previously, that was considered to be 4 bytes wide, but now it is considered to be 8 bytes wide.

All of the offsets involve ICU4XPluralOperands.

File: ffi/diplomat/wasm/lib/api.mjs (hunk 3 above)
Comment (sffc) [reply ffi/diplomat/wasm/lib/api.mjs:78]: (see other thread)

File: ffi/diplomat/wasm/lib/api.mjs (hunk 1 above)
Comment (sffc) [reply ffi/diplomat/wasm/lib/api.mjs:807]: (see other thread)

**Comment by sffc** [comment]
Well, it looks like the tests pass.  I don't know how they were passing before; maybe we don't have good coverage of this.  Follow-up: https://github.com/unicode-org/icu4x/issues/1102

**Pull request merged**